### DISCUSSION TITLE
Sans-Redis

### DISCUSSION THREAD

**Discussion by VMannello** [jonaswinkler/paperless-ng#530] [Archived FRs]
Would you be interested in a PR that allowed for a local KV store / message broker? Redis seems reallly heavy handed for running this on devices like a NAS or Pi or honestly in any environment that isn't leveraging distributed workers. Since you're already distributing work within your library you're likely taking a performance hit incorporating redis on these small devices. My proposal is to mock out redis when no url is provided similar to the SQLite approach. Single container for the win.

**Comment by jonaswinkler** [comment]
If it works, why not? I'm not exactly sure how that's supposed to work. Redis is used for two different purposes in paperless:

* Sending tasks from any part of the application to the task queue. Django-Q is used here.
* Sending consumer status messages from the back end to the front end. That's the status on the dashboard and notifications about new documents. Uses Django Channel Layers (https://channels.readthedocs.io/en/stable/topics/channel_layers.html).

And well, I've configured it as a cache for Django as well, but that's entirely optional.

Remember that things such as consumption tasks, the web server, and the scheduler that regularly checks for new mail all run in different processes, so we need some means of communication between these. An In-memory system will not work.

**Reply by jonaswinkler** [reply]
Also, even under heavy load (that's 16 active task workers processing documents), redis by itself doesn't ever go over 4MB ram used.

![image](https://user-images.githubusercontent.com/17569239/107864689-09c84380-6e5f-11eb-8961-3d98a6826ad1.png)
